Bio/PDB/mmtf/mmtfio.py

```diff
@@ -117,19 +119,20 @@ def _save_structure(self, filepath, select):
 
         chain_ids = [chain.get_id() for chain in self.structure.get_chains()]
         biomoltrans = header_dict["biomoltrans"]
-        for key, value in biomoltrans.items():
-            matrix_items = []  # list of 16 items of 4x4 matrix
-            for line in value[
-                1:
-            ]:  # 3 lines for 3x3 rotation and last column translations
-                matrix_items.extend([float(item) for item in line.split()])
-            matrix_items.extend([0.0, 0.0, 0.0, 1.0])
-
-            encoder.set_bio_assembly_trans(
-                bio_assembly_index=key,
-                input_chain_indices=[chain_ids.index(c) for c in value[0]],
-                input_transform=matrix_items,
-            )
+        if biomoltrans is not None:
```

Comment:
This for me is very weird, because of the comment I made above. `header_dict["biomoltrans"]` should simply throw a KeyError if it's not defined, instead of being defined as None. At best, this should be replaced by something like:

```
biomoltrans = header_dict.get("biomoltrans")
if biolmoltrans is not None:
    # do stuff
```

Reply:
Because of using defaultdict, it was originally giving empty string and not a KeyError and we then set it as None. 

So, now asking if it's not None makes sense. I guess this is more secure against future changes though

Reply:
Ok, changing this broke the tests so I need to change them accordingly 

@jamesmkrieger note to self

Reply:
I have restored it to a defaultdict instead

> Ok, changing this broke the tests so I need to change them accordingly
> 
> @jamesmkrieger note to self